File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L27-RemoveElement.java
// https://leetcode.com/problems/remove-element/
class Solution {
    public void swap (int[] nums, int l, int r) {
        int temp = nums[l];
        nums[l] = nums[r];
        nums[r] = temp;
    }
    public int removeElement(int[] nums, int val) {
        int l = 0, r = nums.length - 1, count = 0;
        while (l <= r) {
            if (nums[l] == val && nums[r] == val) {
                r -= 1;
                count += 1;
            } else if (nums[l] == val) {
                swap(nums, l, r);
                l += 1;
                r -= 1;
                count += 1;
            } else {
                l += 1;
            }
        }
        return nums.length - count;
    }
}

File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L27-remove_element.py
# https://leetcode.com/problems/remove-element/
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l = 0
        r = len(nums) - 1
        count = 0
        while l <= r:
            if nums[l] == val and nums[r] == val:
                r -= 1
                count += 1
            elif nums[l] == val:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
                r -= 1
                count += 1
            else:
                l += 1
        return len(nums) - count
